ob1k-core/src/main/java/com/outbrain/ob1k/client/ClientBuilder.java

```diff
@@ -48,17 +45,12 @@
   public ClientBuilder(final Class<T> type) {
     this.type = type;
     this.httpClientBuilder = new HttpClient.Builder();
-    this.syncFilters = new ArrayList<>();
     this.asyncFilters = new ArrayList<>();
     this.streamFilters = new ArrayList<>();
     this.endpointDescriptors = new HashMap<>();
   }
 
   public ClientBuilder<T> addFilter(final ServiceFilter filter) {
```

Comment:
I believe this entire method needs to be changed.
What is the point of having ServiceFilter now?
What is the point in the boolean logic below?